sdk/src/main/scala/io/horizen/account/mempool/MempoolMap.scala

```diff
@@ -291,6 +291,24 @@ class MempoolMap(
     txsMap
   }
 
+  def mempoolTransactionsMapFrom(executable: Boolean, requestedFrom: AddressProposition): TrieMap[SidechainTypes#SCP, TxByNonceMap] = {
+    val txsMap = TrieMap.empty[SidechainTypes#SCP, TxByNonceMap]
+    val mempoolIdsMap = if (executable)
+      executableTxs
+    else
+      nonExecutableTxs
+    for ((from, nonceIdsMap) <- mempoolIdsMap) {
+      if(from.equals(requestedFrom)) {
+        val nonceTxsMap: mutable.TreeMap[BigInteger, SidechainTypes#SCAT] = new mutable.TreeMap[BigInteger, SidechainTypes#SCAT]()
+        for ((txNonce, txId) <- nonceIdsMap) {
+          nonceTxsMap.put(txNonce, getTransaction(txId).get)
```

Comment:
I wonder if the function transforming EthereumTransaction to TxPoolTransaction cannot be passed here, to avoid iterating twice on the map. For txs belonging to a single account it is no much of a difference but when all the mempool txs are request it may be important.

Reply:
Yes, good point! I proceed to do an extensive refactor on these methods:
- declaring the map with the TxPoolTransactions in the `MempoolMap` class
- using Scala object as return objects of the RPC methods
in this way no double iterations are needed to cast the object from Scala to Java or to declare the correct object for the transaction data.

The same was done for the `txpool_inspect` method moving the response generation in the `MempoolMap` class
Unit tests updated as well